diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index f2d391a9..c8ae36be 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -346,15 +346,15 @@ if(BUILD_python)
     set(version ${PYTHONLIBS_VERSION_STRING})
 
     STRING( REGEX REPLACE "[^0-9]" "" boost_py_version ${version} )
-    find_package(Boost 1.46 COMPONENTS "python-py${boost_py_version}")
-    set(Boost_PYTHON_FOUND ${Boost_PYTHON-PY${boost_py_version}_FOUND})
+    find_package(Boost 1.46 COMPONENTS "python${boost_py_version}")
+    set(Boost_PYTHON_FOUND ${Boost_PYTHON${boost_py_version}_FOUND})
 
     while(NOT "${version}" STREQUAL "" AND NOT Boost_PYTHON_FOUND)
       STRING( REGEX REPLACE "([0-9.]+).[0-9]+" "\\1" version ${version} )
 
       STRING( REGEX REPLACE "[^0-9]" "" boost_py_version ${version} )
-      find_package(Boost 1.46 COMPONENTS "python-py${boost_py_version}")
-      set(Boost_PYTHON_FOUND ${Boost_PYTHON-PY${boost_py_version}_FOUND})
+      find_package(Boost 1.46 COMPONENTS "python${boost_py_version}")
+      set(Boost_PYTHON_FOUND ${Boost_PYTHON${boost_py_version}_FOUND})
 
       STRING( REGEX MATCHALL "([0-9.]+).[0-9]+" has_more_version ${version} )
       if("${has_more_version}" STREQUAL "")
@@ -412,4 +412,4 @@ if(BUILD_docs)
 endif()
 
 include_directories(${Caffe_INCLUDE_DIRS})
-link_directories(${Caffe_LINKER_LIBS})
\ No newline at end of file
+link_directories(${Caffe_LINKER_LIBS})
