diff --git a/cmake/Misc.cmake b/cmake/Misc.cmake
index fcb24647..950d3eb6 100644
--- a/cmake/Misc.cmake
+++ b/cmake/Misc.cmake
@@ -28,10 +28,6 @@ if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
   set(CMAKE_INSTALL_PREFIX "${PROJECT_BINARY_DIR}/install" CACHE PATH "Default install path" FORCE)
 endif()
 
-# ---[ RPATH settings
-set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE CACHE BOOLEAN "Use link paths for shared library rpath")
-set(CMAKE_MACOSX_RPATH TRUE)
-
 list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
      ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR} __is_systtem_dir)
 if(${__is_systtem_dir} STREQUAL -1)
 